extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/util/DimensionSchemaUtils.java

```diff
@@ -74,11 +92,33 @@ public static DimensionSchema createDimensionSchema(
         case ARRAY:
           switch (type.getElementType().getType()) {
             case STRING:
-              return new StringDimensionSchema(column, DimensionSchema.MultiValueHandling.ARRAY, null);
+              if (arrayIngestMode == ArrayIngestMode.NONE) {
+                throw InvalidInput.exception(
+                    "String arrays can not be ingested when '%s' is set to '%s'. Either set '%s' in query context "
+                    + "to 'array' to ingest the string array as an array, or set it to 'mvd' to ingest the string array "
+                    + "as MVD (which is legacy behaviour and not recommmended)",
+                    MultiStageQueryContext.CTX_ARRAY_INGEST_MODE,
+                    StringUtils.toLowerCase(arrayIngestMode.name()),
+                    MultiStageQueryContext.CTX_ARRAY_INGEST_MODE
```

Comment:
I don't think we should recommend MVD mode at all here, instead we should always recommend array mode and suggest using `ARRAY_TO_MV` if people want to store things as a MVD.